core/domain/question_domain.py

```diff
@@ -782,19 +783,21 @@ def partial_validate(self):
         if len(set(self.linked_skill_ids)) != len(self.linked_skill_ids):
             raise utils.ValidationError(
                 'linked_skill_ids has duplicate skill ids')
-
-        if not (isinstance(self.inapplicable_misconception_ids, list) and (
+        inapplicable_skill_misconception_ids_is_list = isinstance(
+            self.inapplicable_skill_misconception_ids, list)
+        if not (inapplicable_skill_misconception_ids_is_list and (
                 all(isinstance(
                     elem, python_utils.BASESTRING) for elem in (
-                        self.inapplicable_misconception_ids)))):
+                        self.inapplicable_skill_misconception_ids)))):
             raise utils.ValidationError(
-                'Expected inapplicable_misconception_ids to be a list of '
-                'strings, received %s' % self.inapplicable_misconception_ids)
+                'Expected inapplicable_skill_misconception_ids to be a list '
+                'of strings, received %s'
+                % self.inapplicable_skill_misconception_ids)
```

Comment:
Should we add a pattern validation check for `inapplicable_skill_misconception_id` similar to `state.tagged_misconception_id`? (In a separate PR!)